source/server/active_tcp_listener.cc

```diff
@@ -123,6 +123,9 @@ void ActiveTcpSocket::unlink() {
   }
   // Emit logs if a connection is not established.
   if (!connected_) {
+    // Only HCM and tcp proxy filter will populate the server name into stream info.
+    // So the stream info will have no server name at this point.
+    stream_info_->setRequestedServerName(socket_->requestedServerName());
```

Comment:
Instead of having this fallback in case a L4 filter sets this, would it be possible to consolidate where we set this so that filters don't need to be concerned with it? Can we set this somewhere in the TLS stack perhaps? 

cc @lizan @ggreenway

Reply:
I agree; this should be set in a common location, earlier in the process. I think the value is only initially sourced by the tls_inspector, so this could be set in `ActiveTcpListener::newConnection` or somewhere nearby, in `source/server/active_tcp_listener.cc`.

Reply:
tls_inspector sets it on the socket, but listener filters don't have access to the streamInfo, so something has to move it from the socket to the streamIfno. I think `newConnection()`, before bailing out for no-filter-chain-found, is possibly the right place.

Reply:
thanks for the review!

the ActiveTCPSocket own the stream info and passed into listener filter. unfortunately, there is no interface to access that, except we extend the interface https://github.com/envoyproxy/envoy/blob/main/include/envoy/network/filter.h#L264

I also aware @lambdai 's comment https://github.com/envoyproxy/envoy/issues/14922#issuecomment-773606556, it sounds like the SNI may be sniffed by transport socket, but I didn't find it. Maybe have him help to ensure that.

Otherwise, newConnection() sounds like right place.

Reply:
Thank you @soulxu 

I do agree set in newConnection() beats late set in TcpProxy and HCM.
But it's also nice to have it here as well: the newConnection is not always invoked on listening timeout. #14922 describes the scenario and the requirement.


SNI is not sniffed by transport sockets but the tls_inspector: https://github.com/envoyproxy/envoy/blob/main/source/extensions/filters/listener/tls_inspector/tls_inspector.cc#L144

That's what lizan mentions. Although we cannot set the requestservername there because the stream_info is not yet created at that moment

Reply:
> I do agree set in newConnection() beats late set in TcpProxy and HCM.
> But it's also nice to have it here as well: the newConnection is not always invoked on listening timeout. #14922 describes the scenario and the requirement.

I can add it back, thanks for the replay.